tests/update_test_files.py

```diff
@@ -0,0 +1,163 @@
+- Old files in the expected test output that DO NOT exist under `test_outputs` are deleted.
+- Files that are already in the expected test output and have not changed are left alone.
+
+The script prints a log detailing the changes made for each experiment test.
+
+:author: Anastassia Loukina
+:author: Nitin Madnani
+:date: Feburary 2018
+"""
+
+import argparse
+import re
+import sys
+
+from ast import literal_eval as eval
+from filecmp import dircmp
+from importlib.machinery import SourceFileLoader
+from inspect import getmembers, getsourcelines, isfunction
+from os import remove
+from os.path import dirname, join
+from pathlib import Path
+from shutil import copyfile
+
+_MY_PATH = dirname(__file__)
+
+
+def is_skll_excluded_file(filename):
+
+    stem = Path(filename).stem
+    suffix = Path(filename).suffix
+    return suffix == '.model' or \
+        suffix == '.npy'or \
+        stem.endswith('_postprocessing_params') or \
+        stem.endswith('_eval') or \
+        stem.endswith('_eval_short') or \
+        stem.endswith('_confMatrix') or \
+        stem.endswith('_pred_train') or \
+        stem.endswith('_pred_processed') or \
+        stem.endswith('_score_dist')
+
+
+def update_reference_output(source, outputs_dir, skll=False):
+
+    # locate the outputs for the experiment under the outputs directory
+    # and also locate the existing experiment outputs
+    reference_output_path = outputs_dir / source / "output"
+    existing_output_path = Path(_MY_PATH) / "data" / "experiments" / source / "output"
+
+    #  update the report files
+
+    # get a comparison betwen the two directories
+    dir_comparison = dircmp(reference_output_path, existing_output_path)
+
+    # first delete the files that only exist in the existing output directory
+    # since those are likely old files from old versions that we do not need
+    existing_output_only_files = dir_comparison.right_only
+    for file in existing_output_only_files:
+        remove(existing_output_path / file)
+
+    # Next find all the files from the reference path that are either new
+    # or changed compared to the existing path. From these we want to exclude
+    # config JSON files as well as evaluation/prediction/model files for SKLL
+    new_or_changed_files = set(dir_comparison.left_only) - set(dir_comparison.same_files)
+    new_or_changed_files = [f for f in new_or_changed_files if not f.endswith('_rsmtool.json') and not f.endswith('_rsmeval.json')]
+    if skll:
+        new_or_changed_files = [f for f in new_or_changed_files if not is_skll_excluded_file(f)]
+
+    for file in new_or_changed_files:
+        copyfile(reference_output_path / file, existing_output_path / file)
+
+    return list(existing_output_only_files), list(new_or_changed_files)
+
+
+def main():
+    # set up an argument parser
+    parser = argparse.ArgumentParser(prog='update_test_files.py')
+    parser.add_argument('outputs_dir', help="The path to the directory containing reference test outputs")
```

Comment:
Same here - without seeing the code, I would interpret this help message as the location of `tests/data/experiments`.

Reply:
That's what it was already called?

Reply:
I think we agreed above on `updated` and `existing`, right? And these would be `updated` I suppose?